### PR TITLE
(PUP-7493) Record desired mode with leading zeros in the report

### PR DESCRIPTION
**Pull request by joshcooper**
When creating an event for a mode property change, the `desired_value` in the
report is obtained from the `Property#should` method, which calls `unmunge` to
convert the internal representation to its external form. The desired mode
may be an octal string without leading 0's or symbolic form (u+a). Ensure the
latter is padded with 0's so the `desired_value` matches the `previous_value`.

### CODE REVIEW

File: lib/puppet/util/symbolic_file_mode.rb [hunk 1]
```diff
@@ -19,16 +19,28 @@ def valid_symbolic_mode?(value)
...
+
   def normalize_symbolic_mode(value)
     return nil if value.nil?
 
     # We need to treat integers as octal numbers.
+    #
+    # "A numeric mode is from one to four octal digits (0-7), derived by adding
+    # up the bits with values 4, 2, and 1. Omitted digits are assumed to be
+    # leading zeros."
     if value.is_a? Numeric then
```
Comment (GabrielNagy) [inline lib/puppet/util/symbolic_file_mode.rb:38]: Can also drop the `then` here

File: lib/puppet/util/symbolic_file_mode.rb [hunk 2]
```diff
@@ -19,16 +19,28 @@ def valid_symbolic_mode?(value)
...
-      return value.to_s(8)
-    elsif value =~ /^0?[0-7]{1,4}$/ then
-      return value.to_i(8).to_s(8)
+      value.to_s(8)
+    elsif value =~ /^0?[0-7]{1,4}$/
+      value.to_i(8).to_s(8) # this seems odd
```
Comment (GabrielNagy) [inline lib/puppet/util/symbolic_file_mode.rb:41]: Not sure what purpose did `to_i(8)` serve... maybe to ensure the returned output doesn't contain a leading 0 (if `value` is something like `0644`?

**Comment by joshcooper** [comment]
jenkins please test this with servertests

**Comment by joshcooper** [comment]
jenkins please test this with servertests

File: lib/puppet/util/symbolic_file_mode.rb (hunk 2 above)
Comment (joshcooper) [reply lib/puppet/util/symbolic_file_mode.rb:41]: yep, I forgot to update the comment, I'll do that after tests pass.

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**